src/main/java/org/opensearch/flowframework/workflow/AbstractRegisterLocalModelStep.java

```diff
@@ -90,68 +84,65 @@ public CompletableFuture<WorkflowData> execute(
-                requiredKeys,
-                optionalKeys,
+                getRequiredKeys(),
+                getOptionalKeys(),
                 currentNodeInputs,
                 outputs,
                 previousNodeInputs
             );
 
+            // Extract common fields of OS provided text-embedding, sparse encoding and custom models
             String modelName = (String) inputs.get(NAME_FIELD);
             String modelVersion = (String) inputs.get(VERSION_FIELD);
-            String description = (String) inputs.get(DESCRIPTION_FIELD);
-            MLModelFormat modelFormat = MLModelFormat.from((String) inputs.get(MODEL_FORMAT));
-            String modelGroupId = (String) inputs.get(MODEL_GROUP_ID);
+            String modelFormat = (String) inputs.get(MODEL_FORMAT);
+
+            // Extract non-common fields
+            String functionName = (String) inputs.get(FUNCTION_NAME);
             String modelContentHashValue = (String) inputs.get(MODEL_CONTENT_HASH_VALUE);
+            String url = (String) inputs.get(URL);
             String modelType = (String) inputs.get(MODEL_TYPE);
             String embeddingDimension = (String) inputs.get(EMBEDDING_DIMENSION);
-            FrameworkType frameworkType = FrameworkType.from((String) inputs.get(FRAMEWORK_TYPE));
+            String frameworkType = (String) inputs.get(FRAMEWORK_TYPE);
+
+            // Extract optional fields
+            String description = (String) inputs.get(DESCRIPTION_FIELD);
+            String modelGroupId = (String) inputs.get(MODEL_GROUP_ID);
             String allConfig = (String) inputs.get(ALL_CONFIG);
-            String url = (String) inputs.get(URL);
-            String functionName = (String) inputs.get(FUNCTION_NAME);
             final Boolean deploy = (Boolean) inputs.get(DEPLOY_FIELD);
 
-            // Create Model configuration
-            TextEmbeddingModelConfigBuilder modelConfigBuilder = TextEmbeddingModelConfig.builder()
-                .modelType(modelType)
-                .embeddingDimension(Integer.valueOf(embeddingDimension))
-                .frameworkType(frameworkType);
-            if (allConfig != null) {
-                modelConfigBuilder.allConfig(allConfig);
-            }
-            MLModelConfig modelConfig = modelConfigBuilder.build();
-
-            // Create register local model input
+            // Build register model input
             MLRegisterModelInputBuilder mlInputBuilder = MLRegisterModelInput.builder()
                 .modelName(modelName)
                 .version(modelVersion)
-                .modelFormat(modelFormat)
-                .hashValue(modelContentHashValue)
-                .modelConfig(modelConfig)
-                .url(url);
+                .modelFormat(MLModelFormat.from(modelFormat));
+
+            if (functionName != null) {
+                mlInputBuilder.functionName(FunctionName.from(functionName));
```

Comment:
Is this the key feature that defines which model we're using?  

We might want to hardcode this per-step and not collect it from the user.   See https://github.com/opensearch-project/flow-framework/issues/341 for where the "remote" function name actually triggers a synchronous code flow.

I see there's a `Function.SPARSE_ENCODING` which probably applies to that model so that one should probably be hardcoded..  Not sure how this applies to the other models (or can they use any function)?

Reply:
The Sparse embedding models also support a `Function.SPARSE_TOKENIZE` value for the function name, it really depends on the model config that is provided with the model. Since there's multiple options, I would rather we have the user provide this

Reply:
Huh. The docs are confusing.  It says `function_name` is a required input of either SPARSE_ENCODING or SPARSE_TOKENIZE, but then the examples don't use that for opensearch pre-trained, or custom model.

So we might want to have some way of validating that input just for the sparse model.... but not hardcoded since it's still a user's choice.